pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/AbstractJavaRulechainRule.java

```diff
@@ -18,32 +19,25 @@
  */
 public abstract class AbstractJavaRulechainRule extends AbstractJavaRule {
```

Comment:
Maybe we can get rid of this class eventually?

Reply:
Probably yes. But the fact that rules also implement the visitor interface is a problem for that, as the default visit method traverses the tree, which we want to avoid.